### PR TITLE
Fabex

### PR DESCRIPTION
**Pull request by vadimInshakov**
The aim of this lab is to create a flexible block explorer written in Go.

### CODE REVIEW

**Comment by arsulegai** [comment]
How would this compliment https://github.com/hyperledger/blockchain-explorer , would this rather be a cli for HLF?

**Comment by lehors** [comment]
I removed "project" from the name because this label is reserved for official HLP projects approved by the TSC.
Otherwise, I agree with @arsulegai, it would be good to explain how this differs from HLP Explorer, what the motivation to do something different is.
Thanks.

**Comment by vadimInshakov** [comment]
@lehors @arsulegai the main difference is: 
1. GRPC API, not UI for communication;
2. language in which it's written;
3. flexible db interface (2 db's implementations available right now)

It's not a replacement for Hyperledger Explorer, but just another option.

**Comment by tkuhrt** [comment]
I am assuming that this is specifically for Hyperledger Fabric, but the proposal does not mention that. Instead `flexible block explorer written in Go`. Can you update the proposal to reflect that (unless of course you are truly intending for this to be a flexible block explorer for multiple blockchain projects)?

**Comment by vadimInshakov** [comment]
@lehors thanks for the comments. I added a little more information and left some links in the proposal.

**Comment by tkuhrt** [comment]
The existing repository does not have DCO sign-off. Assuming this lab is approved, we will not be able to directly transfer the repository as is. You will either need to modify the commits to include DCO sign off in the existing repository or you will need to bring the code over manually.

**Comment by vadimInshakov** [comment]
@trbs thanks, you absolutely right. I changed description.

**Comment by vadimInshakov** [comment]
> The existing repository does not have DCO sign-off. Assuming this lab is approved, we will not be able to directly transfer the repository as is. You will either need to modify the commits to include DCO sign off in the existing repository or you will need to bring the code over manually.

I need to rewrite all commits?

**Comment by tkuhrt** [comment]
@vadimInshakov If you wish to keep the history, you will need to rewrite the commits in your existing repo. Otherwise, we will need to bring it over to hyperledger-labs organization as a single commit with DCO sign-off

**Comment by vadimInshakov** [comment]
> @vadimInshakov If you wish to keep the history, you will need to rewrite the commits in your existing repo. Otherwise, we will need to bring it over to hyperledger-labs organization as a single commit with DCO sign-off

Ok, done. I rewrote all commits.

**Comment by vadimInshakov** [comment]
> I hate links using "here" as the anchor but will pinch my nose and won't ask for this to be changed. :-)

No problem, fixed :)

**Comment by lehors** [comment]
Thanks! :-)

**Comment by tkuhrt** [comment]
@C0rWin Can you confirm sponsorship?

**Comment by C0rWin** [comment]
> @C0rWin Can you confirm sponsorship?

yes, I do, in fact @vadimInshakov confirmed with me prior to submission. 

> * GRPC API, not UI for communication;
> 
> * language in which it's written;
> 
> * flexible db interface (2 db's implementations available right now)

Let me probably clarify a bit on the points mentioned here, this is an SDK which augments the ledger and index it providing set of additional APIs for comprehensive queries. 

Actually I love the fact that this is has been done as stand alone third party library without trying to change Fabric internal. I think this is might be of practical use in many places.

File: labs/fabex.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Lab Name
+Fabex
+
+# Short Description
+The aim of this lab is to create a flexible block explorer for Hyperledger Fabric written in Go.
```
Comment (C0rWin) [inline labs/fabex.md:5]: I would rephrase it a bit, how about?

"The aim of this library is to provide microservice which augments and index the ledger of Hyperledger Fabric and to provide additional set of complex queries APIs to extract provenance information out of the ledger."

Comment (vadimInshakov) [reply labs/fabex.md:5]: I think, it's right.

File: labs/fabex.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Scope of Lab
+Fabex aims to contribute to the Hyperledger Fabric audit ecosystem. It provides GRPC API and extensible database storage interface (MongoDB and CassandraDB implementations are now available). 
+This tool can be used as CLI or microservice. Simple API documented in [fabex.proto](https://github.com/VadimInshakov/fabex/blob/master/proto/fabex.proto).
```
Comment (C0rWin) [inline labs/fabex.md:9]: Library exposes set of GRPC APIs to execute provenance queries over ledger data, moreover library provides an abstract DB interface allowing to integrate different DB engines, whereas out of box supported:

* MongoDB
* CassandraDB

The library could be integrated and deployed as a microservice, at the same time could be compiled into stand-alone command line tool.

Comment (vadimInshakov) [reply labs/fabex.md]: Ok, I fixed the Scope description.

**Comment by vadimInshakov** [comment]
@tkuhrt @vipinsun @trbs what do you think? Maybe you have any recommendations?

**Comment by ryjones** [comment]
@tkuhrt yes - once this is merged, @vadimInshakov, please transfer the repo to me (@ryjones) and I will put it in place.

**Comment by ryjones** [comment]
@vadimInshakov I need you to transfer ownership - this is in settings for the repo, at the bottom. Thanks.

**Comment by vadimInshakov** [comment]
@ryjones done, thanks.
